app/uk/gov/hmrc/gform/formtemplate/RepeatingComponentService.scala

```diff
@@ -81,23 +81,42 @@ object RepeatingComponentService {
     }
 
     section.copy(
-      title = buildText(Some(section.title), index, section.fieldToTrack.get, data).getOrElse(""),
-      shortName = buildText(section.shortName, index, section.fieldToTrack.get, data),
+      title = buildText(Some(section.title), index, data).getOrElse(""),
+      shortName = buildText(section.shortName, index, data),
       fields = section.fields.map(copyField)
     )
   }
 
-  private def buildText(template: Option[String], index: Int, fieldToTrack: VariableInContext,
-    data: Map[String, String]): Option[String] = {
+  private def buildText(template: Option[String], index: Int, data: Map[String, String]): Option[String] = {
+
+    def evaluateTextExpression(str: String) = {
+      val field = str.replaceFirst("""\$\{""", "").replaceFirst("""\}""", "")
+      if (field.startsWith("n_")) {
+        if (index == 1) {
+          val fieldName = field.replaceFirst("n_", "")
+          data.getOrElse(fieldName, "")
```

Comment:
`data.getOrElse(fieldName, "")` - this worries me. We result in default value when there is a missing field. In my opinion this should be exceptional situation when field can't be resolved and reported as an Exception. Let's discuss it. @lutnos